File: xinya-backend/src/main/java/com/xinya/dtx/patient/dto/PatientDto.java
package com.xinya.dtx.patient.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 患者基础信息 DTO（与 API 文档 PatientDto 对齐）
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PatientDto {

    private String id;

    private String name;

    private String stage;

    private Integer psychEnergy;

    private Integer treeLevel;

    private String admissionDate;

    private String roomNumber;

    private String createdAt;

    private String updatedAt;
}

File: xinya-client/app/src/main/java/com/xinya/dtx/core/network/dto/PatientDto.kt
package com.xinya.dtx.core.network.dto

import com.google.gson.annotations.SerializedName

/**
 * 患者信息DTO
 */
data class PatientDto(
    val id: String,
    val name: String,
    val stage: String,
    @SerializedName("psych_energy")
    val psychEnergy: Int,
    @SerializedName("tree_level")
    val treeLevel: Int,
    @SerializedName("admission_date")
    val admissionDate: String,
    @SerializedName("room_number")
    val roomNumber: String?
)

/**
 * 创建患者请求
 */
data class CreatePatientRequest(
    val name: String,
    @SerializedName("room_number")
    val roomNumber: String?,
    @SerializedName("admission_date")
    val admissionDate: String
)

/**
 * 更新患者阶段请求
 */
data class UpdateStageRequest(
    @SerializedName("patient_id")
    val patientId: String,
    @SerializedName("target_stage")
    val targetStage: String
)
